### ISSUE TITLE
RotateLeft and RotateRight actions: should rotate about viewport centre/center when invoked from the toolbar or menu

### ISSUE THREAD

**Issue by achadwick** [mypaint/mypaint#555]
The canvas rotation commands rotate the viewport around the mouse pointer when they are invoked from the keyboard, which is correct. However when they are invoked from the menus or from the toolbar, the mouse pointer is still used as the rotation centre when it would be better to rotate the canvas around the viewport centre instead.

Is there any way of distinguishing one kind of invocation from another, or do we need separate commands?

Fallout from my looking closely at #553.

**Comment by odysseywestra** [comment]
@briend @jplloyd This is probably should be another issue that should address during the beta phase of MyPaint.

**Comment by jplloyd** [comment]
**TLDR:** We'll fix this for the toolbar, but not the menu.

This is ttrickier than one might think (and definitely trickier than it should be...).
One way to solve this is to use separate GtkAction:s which borrow labels and tooltips from the corresponding original rotation actions, then those could be used from the menu and toolbar (by the way, the default settings for toolbar visibility actually hides these controls, so new users won't even know they _can_ be accessed from the toolbar).

There is a problem with that solution: using a separate action in the _menu_ means that the accelerator for the standard cursor-centered rotation will not show up beside it. If there is no way of providing some piece of data to differentiate the sources of the actions, we'll either have to live with:

**a)** The rotation being centered around the mouse when invoked from the _menu_ (not toolbar, the solution works fine for the toolbar), or
**b)** The accelerator (keyboard shortcut) not showing up besides the rotation action in the menu.

Seeing that the canvas rotation actions from the _menu_ are located **4 layers deep** (Menu -> View -> Adjust View -> Rotate XY), I can't think of a scenario where anyone would realistically use those for anything but reminding yourself of the keyboard shortcut. Even for those who only have the use of one hand, I can't imagine that using the toolbar or a single-key accelerator is not simply better in every respsect.

In short, I favour, and will implement **a)**. If someone knows a way to have the same action work in two different ways depending on _how_ it is activated, that would be preferable of course.

**Comment by jplloyd** [comment]
Toolbar actions now rotate and zoom based on the center, as of: 3b4a6da3806c2b6b9f7a6e754d2a855a3381607a

**Comment by odysseywestra** [comment]
Can confirm it works. Thanks!
